exercises/hangman/Example.cs

```diff
@@ -1,85 +1,91 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
 using System.Linq;
-
-public delegate void HangmanChangedEventHandler(object sender, HangmanState state);
-
+using System.Reactive;
+using System.Reactive.Subjects;
+
+/// <summary>
+/// Returned state is <see href="https://blog.ndepend.com/c-sharp-immutable-types-understanding-attraction/">immutable</see>,
+/// to help the students implement the right solution, escaping the trap of <see cref="RemainingGuesses">RemainingGuesses</see> be the same on
+/// all observed elements.
+/// </summary>
 public class HangmanState
 {
-    public HangmanStatus Status { get; set; }
-    public int RemainingGuesses { get; set; }
-    public string MaskedWord { get; set; }
-    public HashSet<char> Guesses { get; set; }
+    public string MaskedWord { get; }
+    public ImmutableHashSet<char> GuessedChars { get; }
+    public int RemainingGuesses { get; }
+
+    public HangmanState(string maskedWord, ImmutableHashSet<char> guessedChars, int remainingGuesses)
+    {
+        MaskedWord = maskedWord;
+        GuessedChars = guessedChars;
+        RemainingGuesses = remainingGuesses;
+    }
 }
 
-public enum HangmanStatus
+/// <summary>
+/// Notifies the observers, that the game is lost, because of too many guesses.
+/// </summary>
+public class GameFailedException : Exception
 {
-    Busy,
-    Win,
-    Lose
 }
 
-public class HangmanGame
+public class Hangman
 {
-    private const int NumberOfAllowedGuesses = 9;
-    private const char UnguessedCharacterPlaceHolder = '_';
-
-    private readonly string word;
-    private readonly HangmanState state;
+    public IObservable<HangmanState> StateObservable { get; }
+    public IObserver<char> GuessObserver { get; }
+    private const char HidingChar = '_';
+    private const int MaxGuessCount = 9;
 
-    public HangmanGame(string word)
+    public Hangman(string word)
     {
-        this.word = word;
-
-        state = new HangmanState
+        // until Multiple_player_sees_the_same_game_already_started, the tests pass with the following solution> 
```

Comment:
This comment is very useful to me as a reviewer, but less useful to student I think. Could you remove this comment?